### ISSUE TITLE
RegExp: case-insensitive matching not compatible with V8/JSC

### ISSUE THREAD

**Issue by mathiasbynens** [Microsoft/ChakraCore#1136]
Chakra returns `false` for each of these tests, while V8 and JavaScriptCore return `true` for all of them:

```js
/\u01F1/i.test('\u01F3')
/\u0345/i.test('\u03B9')
/\u037F/i.test('\u03F3')
/\u0528/i.test('\u0529')
/\u052A/i.test('\u052B')
/\u052C/i.test('\u052D')
/\u052E/i.test('\u052F')
/\uA698/i.test('\uA699')
/\uA69A/i.test('\uA69B')
/\uA796/i.test('\uA797')
/\uA798/i.test('\uA799')
/\uA79A/i.test('\uA79B')
/\uA79C/i.test('\uA79D')
/\uA79E/i.test('\uA79F')
/\uA7AB/i.test('\u025C')
/\uA7AC/i.test('\u0261')
/\uA7AD/i.test('\u026C')
/\uA7B0/i.test('\u029E')
/\uA7B1/i.test('\u0287')
```

https://mathiasbynens.be/demo/regex-i

(See also: [similar SpiderMonkey bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1280046).)

**Comment by dilijev** [comment]
@bterlson for context in case there is specific text in the spec for cases like this
@digitalinfinity who was looking into unicode exceptions for cross-plat

I wonder whether the behavior has something to do with unicode text APIs on Windows in this case and whether we need to do something specific to comply with spec or cross-platform expectations.

**Comment by digitalinfinity** [comment]
Assigning to @akroshg to investigate

**Comment by obastemur** [comment]
This looks like a duplicate of https://github.com/Microsoft/ChakraCore/issues/517 ?

**Comment by dilijev** [comment]
Looks like it. I say we dupe that one and keep this new one since it was filed by a community member, has more cases to test, Gorkem is no longer on our team, and this one has been assigned.

**Comment by dilijev** [comment]
Closed as duplicate of #517
